app/middleware/parameter-validation.ts

```diff
@@ -30,6 +34,46 @@ function validateLinkTypeParameter(req: HarmonyRequest): void {
   }
 }
 
+/**
+ * Validate that the bucket name provided is in the same AWS region as the given region
+ *
+ * @param bucketName - The name of the s3 bucket
+ * @param region - The name of the aws region
+ */
+async function validateBucketIsInRegion(bucketName: string, region: string): Promise<void> {
+  try {
+    const bucketRegion = await defaultObjectStore().getBucketRegion(bucketName);
+    if (bucketRegion != region) {
+      throw new RequestValidationError(`Destination bucket '${bucketName}' must be in the '${region}' region, but was in '${bucketRegion}'.`);
+    }
+  } catch (e) { 
+    if (e.name === 'NoSuchBucket') {
+      throw new RequestValidationError(`The specified bucket '${bucketName}' does not exist.`);
+    } 
+    throw e;
+  }
+}
+
+/**
+ * Validate that the value provided for the `destinationUrl` parameter is an `s3` url in the format of `s3://<bucket>/<path>` is in the same AWS region
+ *
+ * @param req - The client request
+ */
+async function validateDestinationUrlParameter(req: HarmonyRequest): Promise<void> {
+  const keys = keysToLowerCase(req.query);
+  const destUrl = keys.destinationurl?.toLowerCase();
+  if (destUrl) {
+    if (!destUrl.startsWith('s3://')) {
+      throw new RequestValidationError(`Invalid destinationUrl '${destUrl}' must start with s3://`);
+    }
+    const bucketName = destUrl.substring(5).split('/')[0];
```

Comment:
I'm assuming it is not possible for this to throw an exception? Otherwise I think an internal error would go back to the user.

Reply:
Yes, the check above should prevent any exception from being thrown in this call.